x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list_page.tsx

```diff
@@ -5,20 +5,29 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import { ACTION_HISTORY } from '../../../../app/translations';
 import { AdministrationListPage } from '../../../components/administration_list_page';
 import { ResponseActionsLog } from '../../../components/endpoint_response_actions_list/response_actions_log';
 import { UX_MESSAGES } from '../../../components/endpoint_response_actions_list/translations';
 
 export const ResponseActionsListPage = () => {
+  const [hideHeader, setHideHeader] = useState(true);
+  const setIsDataCallback = useCallback((isData: boolean) => {
+    setHideHeader(!isData);
+  }, []);
   return (
     <AdministrationListPage
       data-test-subj="responseActionsPage"
       title={ACTION_HISTORY}
       subtitle={UX_MESSAGES.pageSubTitle}
+      hideHeader={hideHeader}
     >
-      <ResponseActionsLog showHostNames={true} isFlyout={false} />
+      <ResponseActionsLog
+        showHostNames={true}
+        isFlyout={false}
+        setIsDataCallback={setIsDataCallback}
```

Comment:
We need this for hideHeader prop above as we need to know if there is data at all or not before show the header